src/event.cc

```diff
@@ -258,21 +261,26 @@ EventScheduler::clean()
 }
 
 void
-EventScheduler::dump(Packable *out)
+EventScheduler::dump(std::ostream &yaml)
 {
     if (last_event_ran)
-        out->appendf("Last event to run: %s\n\n", last_event_ran);
-
-    out->appendf("%-25s\t%-15s\t%s\t%s\n",
-                 "Operation",
-                 "Next Execution",
-                 "Weight",
-                 "Callback Valid?");
+        yaml << "last event to run: " << last_event_ran << "\n";
 
+    AtMostOnce header("scheduled events:\n");
     for (auto *e = tasks; e; e = e->next) {
-        out->appendf("%-25s\t%0.3f sec\t%5d\t %s\n",
-                     e->name, (e->when ? e->when - current_dtime : 0), e->weight,
-                     (e->arg && e->cbdata) ? cbdataReferenceValid(e->arg) ? "yes" : "no" : "N/A");
+        yaml << header <<
+             "  - operation: " << e->name << '\n' <<
+             "    secs to next execution: " << std::setprecision(3) << std::fixed << (e->when? e->when - current_dtime : 0) << '\n' <<
+             "    weight: " << e->weight << '\n' <<
+             "    callback valid: ";
+        if (e->arg && e->cbdata) {
+            if (cbdataReferenceValid(e->arg))
+                yaml << "yes\n";
+            else
+                yaml << "no\n";
+        } else {
+            yaml << "N/A\n";
```

Comment:
Just like in other reports you have already converted, please do not report fields that are not applicable to an event. It was difficult or even wrong to skip a field when this was a table. This PR is no longer using a table, removing that obstacle.

Reply:
> for me spaces() can go in as-is. What do I need to do to convince you that it's good enough? 

This PR is not the right place to discuss problems with spaces().

> Right now I have a backlog of completed changes

Please do not interpret my commends below as an implicit agreement with the above "completed" characterization of any pending changes.

Creating more and more non-urgent PRs (or code branches, etc.) that heavily depend on some pending issue resolution will naturally grow that backlog, especially when the exact resolution of that issue is unknown and may impact all those changes quite a bit!

BTW, creating lots of PRs that fix a duplicated issue in a new/untested way is likely to face very similar problems.

I cannot stop new PRs, but I do not understand why keep creating all that extra work/overhead.

>  please advise if you have any idea on how I can keep myself unblocked otherwise

My recommendation is to work towards spaces() resolution. If you have free time after that, pick a non-YAML problem that does _not_ depend on spaces() resolution.